### PR TITLE
Add search object and filtering API

### PR DESCRIPTION
**Pull request by robinclart**
⚠️ This PR is a work in progress.

### CODE REVIEW

File: lib/omise/charge.rb [hunk 1]
```diff
@@ -3,12 +3,34 @@
...
 require "omise/transaction"
 
 module Omise
   class Charge < OmiseObject
     self.endpoint = "/charges"
 
+    AVAILABLE_SEARCH_FILTERS = %w[
+      amount
+      authorized
+      auto_capture
```
Comment (spacebra) [inline lib/omise/charge.rb]: changed to capture in #1476

File: lib/omise/charge.rb [hunk 2]
```diff
@@ -3,12 +3,34 @@
...
+      captured
+      captured_at
```
Comment (spacebra) [inline lib/omise/charge.rb]: wanted to keep it till we introduce `created_at` in api next version, should I, should I not? 💭

File: lib/omise/dispute.rb [hunk 3]
```diff
@@ -1,11 +1,23 @@
...
+require "omise/search_scope"
 
 module Omise
   class Dispute < OmiseObject
     self.endpoint = "/disputes"
 
+    AVAILABLE_SEARCH_FILTERS = %w[
+      card_last_digits
+      created
+      reason_code
```
Comment (spacebra) [inline lib/omise/dispute.rb:13]: `reason_code` added in #1476 and it hasn't been merge na ~

File: lib/omise/recipient.rb [hunk 4]
```diff
@@ -1,10 +1,19 @@
 require "omise/object"
 require "omise/bank_account"
+require "omise/search_scope"
 
 module Omise
   class Recipient < OmiseObject
     self.endpoint = "/recipients"
 
+    AVAILABLE_SEARCH_FILTERS = %w[
+      kind
```
Comment (spacebra) [inline lib/omise/recipient.rb]: changed to `type` in #1476

File: lib/omise/charge.rb (hunk 2 above)
Comment (robinclart) [reply lib/omise/charge.rb]: No I can remove it.

**Pull request closed without merging**